oak-search/src/main/java/org/apache/jackrabbit/oak/plugins/index/search/spi/query/FulltextIndexPlanner.java

```diff
@@ -884,7 +884,7 @@ private static boolean isEqualityRestriction(PropertyRestriction pr) {
                 orderEntries.add(IndexDefinition.NATIVE_SORT_ORDER);
             }
             for (PropertyDefinition functionIndex : rule.getFunctionRestrictions()) {
-                if (o.getPropertyName().equals(functionIndex.function)) {
+                if (o.getPropertyName().equals(functionIndex.function) && functionIndex.ordered) {
```

Comment:
Ah, is this a bugfix?

I would probably reverse the order of the condition:
    functionIndex.ordered && o.getPropertyName().equals(functionIndex.function)
because testing a boolean is faster than comparison, but also because of readability of the code... first check if the property is ordered in the index, and then test if it matches.

Reply:
Yes - this was a bugfix. Agreed - will make the ordering change